runtime-testsuite/resources/org/antlr/v4/test/runtime/descriptors/ParserExec/ListLabelsOnRuleRefStartOfAlt.txt

```diff
@@ -0,0 +1,34 @@
+[notes]
+Checks that this compiles; see https://github.com/antlr/antlr4/issues/2016
+
+[type]
+Parser
+
+[grammar]
+grammar Test;
+
+expression
+@after {
+<AssertIsList("$args")>
+}
+    : op=NOT args+=expression
+    | args+=expression (op=AND args+=expression)+
+    | args+=expression (op=OR args+=expression)+
+    | IDENTIFIER
+    ;
+
+AND : 'and' ;
+OR : 'or' ;
+NOT : 'not' ;
+IDENTIFIER : [a-zA-Z_][a-zA-Z0-9_]* ;
+WS : [ \t\r\n]+ -> skip ;
+
```

Comment:
By the way, my sample from https://github.com/antlr/antlr4/pull/3837 looks more succinctly:

```g4
grammar Test;

r
    : args+=r (op=PLUS args+=r)+
    | IDENTIFIER
    ;
PLUS : '+';
IDENTIFIER : [a-zA-Z_][a-zA-Z0-9_]*;
```

Reply:
Ah. I didn't see that one there. I believe I cut and paste something from code you or someone else was proposing earlier to make that one.  Are you suggesting we drop ListLabelsOnRuleRefStartOfAlt.txt because it is redundant? I think it is testing the same thing so maybe we should. Well I guess my checks to type and yours doesn't check the output :)

Reply:
Yes, it's the same, but in more short form.

Reply:
OK but maybe we should test the output  on yours as I think you asked me to do on this test ha ha   I had trouble with the output templates in so I elected to simply check weather the attribute was a list.  Maybe create a PR that adds

```
@after {
<AssertIsList("$r")>
}
```

to yours and then deletes ListLabelsOnRuleRefStartOfAlt.txt?

Reply:
> Ah. I didn't see that one there. I believe I cut and paste something from code you or someone else was proposing earlier to make that one. Are you suggesting we drop ListLabelsOnRuleRefStartOfAlt.txt because it is redundant? I think it is testing the same thing so maybe we should. Well I guess my checks to type and yours doesn't check the output :)

Yes - I think that this code was with the original issue, so I just made the <AssertIsList()> work. Either way is fine by me. Just tell me if you wish to lose the comments generated by <AssertIsList> - or you guys can just edit them out of PR #3848 and merge it. The other code in that PR is required for the test to work and fix the bug that was causing it to fail.